packages/core/utils/serializer.js

```diff
@@ -0,0 +1,61 @@
+function serialize(object) {
+  return JSON.stringify(object, (key, value) => {
+    let serialized = value;
+
+    // If the object has a serialize method, call it
+    if (value && typeof value.serialize === 'function') {
+      serialized = value.serialize();
+    }
+
+    // Add a $$type property with the export specifier for this class, if any.
+    if (
+      value &&
+      typeof value.constructor === 'function' &&
+      value.constructor.__exportSpecifier
+    ) {
+      serialized = Object.assign(
```

Comment:
Not sure about this, but what about using 
```
{
  $$type: value.constructor.__exportSpecifier,
  value: serialized
}
```
it should be faster than Object.assign and in the very unlikely case $$type would already exist in the serialised data it wouldn't override it

Reply:
Yeah that's probably a good idea.

Reply:
Well... I had to restore the Object.assign otherwise it seems to get into a recursive loop. Kept the separate `value` property though